### PR TITLE
CXX-810 Remove WriteConcern fsync property

### PR DESCRIPTION
**Pull request by tessavitabile**

### CODE REVIEW

File: src/mongocxx/write_concern.hpp [hunk 1]
```diff
@@ -52,8 +52,6 @@ class uri;
 ///
 /// @see http://docs.mongodb.org/manual/core/write-concern/
 ///
-/// @todo this interface is terrible -- it's exactly what you get with the c-driver essentially.
```
Comment (tessavitabile) [inline src/mongocxx/write_concern.hpp]: Do you think the interface is terrible? I don't know why it's an issue to have it be so similar to the C driver.

**Comment by acmorrow** [comment]
@ajdavis - Has the C driver not deprecated these interfaces? I'm surprised we were able to use them without seeing a deprecation warning.

File: src/mongocxx/write_concern.hpp (hunk 1 above)
Comment (acmorrow) [inline src/mongocxx/write_concern.hpp]: I don't think there is anything particularly wrong with it. Who wrote the comment?

**Comment by acmorrow** [comment]
I love pure negative line count commits. LGTM

File: src/mongocxx/write_concern.hpp (hunk 1 above)
Comment (tessavitabile) [inline src/mongocxx/write_concern.hpp]: It appears to be Tyler Brock.

Comment (acmorrow) [inline src/mongocxx/write_concern.hpp]: @TylerBrock - any recollection of what you thought was wrong with this one?

Comment (TylerBrock) [inline src/mongocxx/write_concern.hpp]: hey everyone, I don't recall why I thought there was room for improvement, forge ahead!

Comment (acmorrow) [inline src/mongocxx/write_concern.hpp]: Thanks @TylerBrock!

**Comment by acmorrow** [comment]
LGTM

File: src/mongocxx/write_concern.hpp (hunk 1 above)
Comment (samantharitter) [inline src/mongocxx/write_concern.hpp]: Hi Tyler!

**Comment by samantharitter** [comment]
LGTM

**Comment by ajdavis** [comment]
@acmorrow good point, I've opened https://jira.mongodb.org/browse/CDRIVER-1061 to deprecate get/set_fsync.

**Pull request merged**